Source/Core/VideoCommon/BPMemory.h

```diff
@@ -491,6 +491,81 @@ struct fmt::formatter<TevStageCombiner::ColorCombiner>
   template <typename FormatContext>
   auto format(const TevStageCombiner::ColorCombiner& cc, FormatContext& ctx)
   {
+    auto out = ctx.out();
+    if (cc.bias != TevBias::Compare)
+    {
+      // Generate an equation view, simplifying out addition of zero and multiplication by 1
+      // dest = (d (OP) ((1 - c)*a + c*b) + bias) * scale
+
+      // Note that this is simplified with regards to how lerping works; see PixelShaderGen's
+      // WriteTevRegular for more details.
+
+      static constexpr Common::EnumMap<const char*, TevColorArg::Zero> alt_names = {
+          "prev.rgb", "prev.aaa", "c0.rgb",  "c0.aaa",  "c1.rgb", "c1.aaa", "c2.rgb",    "c2.aaa",
+          "tex.rgb",  "tex.aaa",  "ras.rgb", "ras.aaa", "1",      ".5",     "konst.rgb", "0",
+      };
+
+      const bool has_d = cc.d != TevColorArg::Zero;
+      // If c is one, (1 - c) is zero, so (1-c)*a is zero
+      const bool has_a = cc.a != TevColorArg::Zero && cc.c != TevColorArg::One;
+      // If either b or c is zero, b*c is zero
+      const bool has_b = cc.b != TevColorArg::Zero && cc.c != TevColorArg::Zero;
+      const bool has_bias = cc.bias != TevBias::Zero;  // != Compare is already known
+      const bool has_scale = cc.scale != TevScale::Scale1;
+
+      const char op = (cc.op == TevOp::Sub ? '-' : '+');
+
+      if (cc.dest == TevOutput::Prev)
+        out = format_to(out, "dest.rgb = ");
+      else
+        out = format_to(out, "{:n}.rgb = ", cc.dest);
+
+      if (has_scale)
+        out = format_to(out, "(");
+      if (has_d)
+        out = format_to(out, "{}", alt_names[cc.d]);
+      if (has_a || has_b)
+      {
+        if (has_d)
+          out = format_to(out, " {} ", op);
+        else if (cc.op == TevOp::Sub)
+          out = format_to(out, "{}", op);
+      }
+      if (has_a)
+      {
+        if (cc.c == TevColorArg::Zero)
+          out = format_to(out, "{}", alt_names[cc.a]);
+        else if (cc.c == TevColorArg::Half)
+          out = format_to(out, ".5*{}", alt_names[cc.a]);
+        else
+          out = format_to(out, "(1 - {})*{}", alt_names[cc.c], alt_names[cc.a]);
+      }
+      if (has_a && has_b)
+        out = format_to(out, " {} ", op);
```

Comment:
Yep, this should be hardcoded to `+`.  To put it another way, what is currently implemented is this:

```
dest = (d (OP) ((1 - c)*a (OP) c*b) + bias) * scale
```

but it should be this:

```
dest = (d (OP) ((1 - c)*a + c*b) + bias) * scale
```

Good catch.

Reply:
Actually, no, this is correct; it generates

```
dest = (d (OP) (1 - c)*a (OP) c*b + bias) * scale
```

which is equivalent; it's stripped away a layer of parentheses and distributed the subtraction (if OP is `-`).  I believe I did this to simplify the code, but I should probably add a comment.

Reply:
I think that format is a bit hard for the reader to parse, especially since these equations are never represented like that anywhere else.

What do you think about my suggestion to represent this as just

     dest = (d (OP) lerp(a, b, c) + bias) * scale

when a, b and c are all non-zero/non-one. Would also avoid the need for that extra set of parentheses

Reply:
Yeah, I think that lerp would help.